hooks/pre-checkout

```diff
@@ -74,16 +79,58 @@ export_canva_origin_url() {
   export BUILDKITE_REPO="${origin_url}"
 }
 
+# DEPRECTATED (once app ID and secred are removed, this function can be removed)
+export_legacy_https_repo() {
+  local payload
+  payload=$(jq -n --arg AppID "${GITHUB_APP_ID}" \
+            --arg SecretId "${GITHUB_APP_SECRET}" \
+            '{$AppID, $SecretId}')
+
+  # shellcheck disable=SC2064
+  local aws_version
+  aws_version=$(print_aws_version)
+  if [[ "${aws_version}" == "2" ]]; then
+    # when running aws cli v2, --cli-binary-format raw-in-base64-out is required
+    aws lambda invoke \
+        --region us-east-1 \
+        --function-name github-app-token:stable \
+        --cli-binary-format raw-in-base64-out \
+        --payload "${payload}" \
+        "${OUTFILE}"
+  else
+    aws lambda invoke \
+        --region us-east-1 \
+        --function-name github-app-token:stable \
+        --payload "${payload}" \
+        "${OUTFILE}"
+  fi
+
+  local token
+  token=$(jq -r '.body' <"${OUTFILE}" | jq -r '.token')
+
+  export GITHUB_FETCH_APP_ACCESS_TOKEN="${token}"
+  # Converts repo url from git@github.com:Org/repo to https://x-access-token:token@github.com/Org/repo
+  export CANVA_REPO="https://x-access-token:${token}@github.com/${BUILDKITE_REPO#*@github.com:}"
+  export BUILDKITE_REPO="https://x-access-token:${token}@github.com/${BUILDKITE_REPO#*@github.com:}"
+}
+
 main() {
   if [[ -n "${SSM_PARAM:-}" ]]; then
     export_canva_origin_url
   else
     log_info "ssm param not configured skipping"
   fi
   if [[ "${USE_HTTPS_REPO_URL:-false}" = "true" ]]; then
-    # Converts repo url from git@github.com:Org/repo (SSH) to https://github.com/Org/repo
-    export CANVA_REPO="https://github.com/${BUILDKITE_REPO#*@github.com:}"
-    export BUILDKITE_REPO="https://github.com/${BUILDKITE_REPO#*@github.com:}"
+
+    # DEPRECATED (app ID and secret will be removed, only export plain HTTPS urls from else case in the future)
+    if [[ -n "${GITHUB_APP_ID:-}" ]] && [[ -n "${GITHUB_APP_SECRET:-}" ]]; then
+      export_legacy_https_repo
```

Comment:
this check should be outside the previous if statement. otherwise we will use github app id and secret only if setting the `use_https` var to true.

I would say something like
```
if [[ -n "${GITHUB_APP...}"]]; then
  export_legacy
elif [[ "${USE_HTTPS_REPO_URL:-false}" = "true" ]]; then
  export http url
else
  export ssh url
fi
```
this way we make sure all old pipelines that haven't been updated will still use the github app logic, and only when removing it and setting the `use_https` var they will run the new logic.

WDYT?

Reply:
It was intentional. I would still update the plugin parameter to add `use_https_repo_url: true`, but also leave the app id and secret in there. So I would still touch the code where we can not update, it's just that there would be no functional change in how it operates.

My idea was that it still will be clear that we rewrite the URL to HTTPS and be on par with everything else. E.g. in places where we use SSH, there is now `use_https_repo_url: false` where previously there where no parameters at all.